contractsV5/KyberHintHandler.sol

```diff
@@ -223,85 +184,127 @@ contract KyberHintParser is IKyberHint, Utils {
+            reservesHint.tradeType = TradeType.MaskIn;
+            (indexToContinueFrom) = decodeReservesFromHint(false, hint, reservesHint, indexToContinueFrom);
             decodeOperation(hint, tradeHint, indexToContinueFrom, isTokenToEth);
-        } else if (opcodeHash == keccak256(parseOpcode(TradeType.MaskOut))) {
-            reserves.tradeType = TradeType.MaskOut;
-            (reserves.addresses, reserves.splitValuesBps, indexToContinueFrom) = decodeReservesFromHint(false, hint, indexToContinueFrom);
+        } else if (opcodeHash == MASK_OUT_KECCAK) {
+            reservesHint.tradeType = TradeType.MaskOut;
+            (indexToContinueFrom) = decodeReservesFromHint(false, hint, reservesHint, indexToContinueFrom);
             decodeOperation(hint, tradeHint, indexToContinueFrom, isTokenToEth);
-        } else if (opcodeHash == keccak256(parseOpcode(TradeType.Split))) {
-            reserves.tradeType = TradeType.Split;
-            (reserves.addresses, reserves.splitValuesBps, indexToContinueFrom) = decodeReservesFromHint(true, hint, indexToContinueFrom);
+        } else if (opcodeHash == SPLIT_TRADE_KECCAK) {
+            reservesHint.tradeType = TradeType.Split;
+            (indexToContinueFrom) = decodeReservesFromHint(true, hint, reservesHint, indexToContinueFrom);
             decodeOperation(hint, tradeHint, indexToContinueFrom, isTokenToEth);
-        } else {
-            revert("Invalid hint opcode");
+        } else {            
+            reservesHint.invalidHint = true;
         }
     }
 
     function decodeReservesFromHint(
         bool isTokenSplit,
         bytes memory hint,
+        ReservesHint memory reservesHint,
         uint indexToContinueFrom
     )
         internal
         view
         returns (
-            IKyberReserve[] memory,
-            uint[] memory,
             uint
         )
     {
        uint reservesLength = hint.toUint8(indexToContinueFrom);
        uint bpsSoFar;
        uint[] memory splitValuesBps;
-
        IKyberReserve[] memory addresses = new IKyberReserve[](reservesLength);
+
        if (isTokenSplit) {
            splitValuesBps = new uint[](reservesLength);
        } else {
            splitValuesBps = new uint[](1);
            splitValuesBps[0] = BPS;
        }
-       indexToContinueFrom++;
 
-       for (uint i = 0; i < reservesLength; i++) {           
+       indexToContinueFrom++;      
+
+       for (uint i = 0; i < reservesLength; i++) {
            addresses[i] = IKyberReserve(convertReserveIdToAddress(hint.slice(indexToContinueFrom, RESERVE_ID_LENGTH).toBytes8(0)));
+
            indexToContinueFrom += RESERVE_ID_LENGTH;
+
            if (isTokenSplit) {
                splitValuesBps[i] = uint(hint.toUint16(indexToContinueFrom));
                bpsSoFar += splitValuesBps[i];
                indexToContinueFrom += 2;
            }
        }
+
        require((bpsSoFar == BPS) || (bpsSoFar == 0), "bps do not sum to 10000");
```

Comment:
shorter message.
maybe BPS > 10000
and add indexToContinueFrom
to get more helpful message.

Reply:
Done